core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -736,6 +732,12 @@ public synchronized boolean doLeave(Address leaver) throws Exception {
    }
 
    public synchronized void startQueuedRebalance() {
+      // We cannot start rebalance until queued CR is complete
+      if (queuedConflictTopology != null) {
+         log.tracef("Postponing rebalance for cache %s as conflict resolution is in progress", cacheName);
+         return;
```

Comment:
I think we need `onConflictResolutionEnd` as different actions are taken by the two strategies.

Reply:
I think the actions are pretty much the same now, because in `PreferConsistencyStrategy` the availability mode should already be available and the list of members should be correct. In fact, if `updateMembersAndRebalance` has to do more than just queue the rebalance there, I'd argue that the `PreferAvailabilityStrategy` should be modified to do the same.